### PR TITLE
IPT Unit Tests

### LINKED ISSUE #116: Unit tests for pallet_ipt
- [ ] Write tests for all functions in pallet_ipt

### PR DESCRIPTION
**Pull request by S0raWasTaken**
Add Ipt unit tests
closes #116

### CODE REVIEW

**Review by S0raWasTaken — COMMENTED** [review]
@arrudagates I have an issue with `mock.rs`. It would be a great help if you could look into it

**Review by arrudagates — COMMENTED** [review]
Looks good, just missing the vote tests now.

File: INV4/pallet-ipt/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+        assert_ok!(Ipt::burn(
+            Origin::signed(multi_account_id::<Runtime, IptId>(0, None)),
+            0,
+            ExistentialDeposit::get() * 2 + 1001, /*Burning BOB's tokens*/
+            BOB
+        ));
+
+        // Call won't be rechecked until ALICE or VADER tries voting again,
+        // this should work even if they are already voters.
+        assert_ok!(Ipt::vote_multisig(Origin::signed(ALICE), 0, call_hash)); // fails: NotEnoughAmount
```
Comment (arrudagates) [inline INV4/pallet-ipt/src/tests.rs:592]:
```suggestion
        //  assert_ok!(Ipt::vote_multisig(Origin::signed(ALICE), 0, call_hash)); // fails: NotEnoughAmount
```
Let's track this in a separate issue, but for now we will force the tests to pass, so they don't block other ongoing tasks

**Review by arrudagates — CHANGES_REQUESTED** [review]

File: INV4/pallet-ipt/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+            ExistentialDeposit::get() * 2 + 1001, /*Burning BOB's tokens*/
+            BOB
+        ));
+
+        // Call won't be rechecked until ALICE or VADER tries voting again,
+        // this should work even if they are already voters.
+        //  assert_ok!(Ipt::vote_multisig(Origin::signed(ALICE), 0, call_hash)); // fails: NotEnoughAmount
+
+        assert_eq!(Multisig::<Runtime>::get((0, call_hash)), None);
+        assert_eq!(Balance::<Runtime>::get(0, BOB), Some(1000));
```
Comment (arrudagates) [inline INV4/pallet-ipt/src/tests.rs:595]:
```suggestion
        // assert_eq!(Multisig::<Runtime>::get((0, call_hash)), None);
        // assert_eq!(Balance::<Runtime>::get(0, BOB), Some(1000));
```

**Pull request merged**